service/arm_templates/security_groups_setup.json

```diff
@@ -82,7 +85,7 @@
               "sourcePortRange": "*",
               "destinationPortRange": "*",
               "sourceAddressPrefix": "[parameters('masterSubnetCidr')]",
-              "destinationAddressPrefix": "[parameters('masterSubnetCidr')]",
+              "destinationAddressPrefix": "[parameters('virtualNetworkCidr')]",
```

Comment:
pods also live inside this network, so allow whole virtual network as destination (pods network always part of virtual network).